contract_isp_invoice/contract.py

```diff
@@ -20,34 +20,36 @@
 #
 ##############################################################################
 
-import sys
+import calendar
+import datetime
 import logging
+import sys
 import time
-import datetime
+
 from openerp.osv import orm, fields
 from openerp.tools.translate import _
-import openerp.addons.decimal_precision as dp
-from openerp.addons.contract_isp.contract import add_months, date_interval
+from openerp.addons.contract_isp.contract import add_months
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
 from openerp import netsvc
 import openerp.exceptions
 
-_logger = logging.getLogger(__name__)
-
+from .invoice import PROCESS_PRORATA, PROCESS_RECURRENT, PROCESS_INITIAL
 
-class res_company(orm.Model):
-    _inherit = "res.company"
+_logger = logging.getLogger(__name__)
 
-    def _days(self, cr, uid, context=None):
-        return tuple([(str(x), str(x)) for x in range(1, 29)])
 
-    _columns = {
-        'invoice_day': fields.selection(_days, 'Invoice day'),
-        'send_email_contract_invoice': fields.boolean('Send invoice by email')
-    }
+def count_months_stupid(from_date, to_date):
+    from_date = from_date.replace(day=1)
+    to_date = to_date.replace(day=1)
+    # Don't even bother making this smart for now
+    count = 0
+    while from_date < to_date:
+        count += 1
+        from_date = (
+            from_date + datetime.timedelta(days=32)
+        ).replace(day=1)
```

Comment:
oh boy... 

dateutil is a standard openerp dependency, and it provides datetutil.relativedelta... 

https://labix.org/python-dateutil#head-ba5ffd4df8111d1b83fc194b97ebecf837add454